src/js/media-captions-list.js

```diff
@@ -0,0 +1,137 @@
+
+
+class MediaCaptionsList extends MediaChromeListbox {
+  #subs = [];
+  #caps = [];
+
+  static get observedAttributes() {
+    return [
+      ...super.observedAttributes,
+      MediaUIAttributes.MEDIA_CAPTIONS_LIST,
+      MediaUIAttributes.MEDIA_CAPTIONS_SHOWING,
+      MediaUIAttributes.MEDIA_SUBTITLES_LIST,
+      MediaUIAttributes.MEDIA_SUBTITLES_SHOWING,
+    ];
+  }
+
+  attributeChangedCallback(attrName, oldValue, newValue) {
+    if (attrName === MediaUIAttributes.MEDIA_SUBTITLES_LIST && oldValue !== newValue) {
+      this.#subs = parseTextTracksStr(this.getAttribute(MediaUIAttributes.MEDIA_SUBTITLES_LIST) ?? '');
+      this.#render();
+
+    } else if (attrName === MediaUIAttributes.MEDIA_CAPTIONS_LIST && oldValue !== newValue) {
+      this.#caps = parseTextTracksStr(this.getAttribute(MediaUIAttributes.MEDIA_CAPTIONS_LIST) ?? '');
+      this.#render();
+
+    } else if (attrName === MediaUIAttributes.MEDIA_SUBTITLES_SHOWING && oldValue !== newValue) {
+      const selectedTrack = parseTextTrackStr(this.getAttribute(MediaUIAttributes.MEDIA_SUBTITLES_SHOWING) ?? '');
+
+      this.#subs.forEach(track => {
+        track.selected = track.language === selectedTrack.language && track.label === selectedTrack.label;
+      });
+      this.#render();
+
+    } else if (attrName === MediaUIAttributes.MEDIA_CAPTIONS_SHOWING && oldValue !== newValue) {
+      const selectedTrack = parseTextTrackStr(this.getAttribute(MediaUIAttributes.MEDIA_CAPTIONS_SHOWING) ?? '');
+
+      this.#caps.forEach(track => {
+        track.selected = track.language === selectedTrack.language && track.label === selectedTrack.label;
+      });
+      this.#render();
+    }
+
+    super.attributeChangedCallback(attrName, oldValue, newValue);
+  }
+
+  connectedCallback() {
+    this.#render();
+
+    this.addEventListener('change', this.#onChange);
+
+    super.connectedCallback();
+  }
+
+  disconnectedCallback() {
+    this.removeEventListener('change', this.#onChange);
+
+    super.disconnectedCallback();
+  }
+
+  #perTypeRender(tracks, type) {
+    tracks.forEach(track => {
+      let option = track.el;
+      let alreadyInDom = true;
+
+      if (!option) {
+        option = document.createElement('media-chrome-listitem');
+        alreadyInDom = false;
+
+        option.value = type + '!' + formatTextTrackObj(track);
+        option.innerHTML = track.label + (track.kind === 'captions' ? ' (cc)' : '');
+      }
+
+
+      if (track.selected) {
+        option.setAttribute('aria-selected', 'true');
+      } else {
+        option.setAttribute('aria-selected', 'false');
+      }
+
+      if (!alreadyInDom) {
+        this.append(option);
+        track.el = option;
+      }
+    });
+  }
+
+  #render() {
+    this.#perTypeRender(this.#subs, 'subs');
+    this.#perTypeRender(this.#caps, 'cc');
+  }
+
+  #onChange(e) {
+    const [newType, selectedOption] = this.selectedOptions[0]?.value?.split('!') ?? [];
```

Comment:
idea(non-blocking): A lot of UIs will have some sort of "default" item to select for lists. In the case of Captions/Subtitles Menus, that would be some kind of "None" option. In the case of (future) Rendition Selection, that would be some kind of "Automatic" option. We may want to anticipate that kind of capability/config use case (which may also mean some mechanism to disable "click etc. to deselect selected item"). Food for thought.

Reply:
You mean like an `off` item? I think that would work but yeah, would require a bit of work. Potentially also need to not unset an option if clicked on a selected option in this case.

Reply:
yup, we'd have to decouple some assumptions here. A "light lift" version would be to have a "baked in" version that can be turned on/off via an attr that we could identify as the "off" item in the logic here to do some "forking paths" on behavior.